src/common/KernelBase.hpp

```diff
@@ -230,6 +230,9 @@ class KernelBase
 #if defined(RAJA_ENABLE_TARGET_OPENMP)
   virtual void runOpenMPTargetVariant(VariantID vid, size_t tune_idx) = 0;
 #endif
+#if defined(RUN_KOKKOS)
+  virtual void runKokkosVariant(VariantID vid, size_t tune_idx) = 0;
```

Comment:
Its probably easier to provide a default implementation here while you're adding kokkos variants than trying to ifdef things in multiple plcaes.